tests/prior_device_tests.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import torch
+from torch.distributions import Beta, Gamma, Normal
+
+from sbi.utils.torchutils import BoxUniform, process_device
+from sbi.utils.user_input_checks_utils import (
+    MultipleIndependent,
+    PytorchReturnTypeWrapper,
+)
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize("device", ["cpu", "mps", "gpu"])
+def test_BoxUniform(device: str):
+    device = process_device(device)
+    low = torch.tensor([0.0])
+    high = torch.tensor([1.0])
+    prior = BoxUniform(low, high)
+    prior.sample((1,))
+    assert prior.device == "cpu"
```

Comment:
Please also test if the sample is on the correct device (and additionally, the log_prob). This currently just checks that the new attribute is changed, but not necessarily the tensors

Reply:
Generally, for the tests, the sample and log_prob are the "most" important in sbi, so its fine to check these (this would also allow for a single test which can be run for all priors.